src/server/object_services/object_server.js

```diff
@@ -66,14 +68,27 @@ async function create_object_upload(req) {
         content_type: req.rpc_params.content_type ||
             mime.getType(req.rpc_params.key) ||
             'application/octet-stream',
-        upload_started: obj_id,
-        upload_size: 0,
         tagging: req.rpc_params.tagging,
         encryption
     };
 
-    const lock_settings = config.WORM_ENABLED ? calc_retention(req) : undefined;
-    if (lock_settings) info.lock_settings = lock_settings;
+    if (req.rpc_params.complete_upload) {
+        if (!req.rpc_params.size || req.rpc_params.size < 0) {
+            throw new RpcError('INVALID_REQUEST', 'valid size must be provided when using complete_upload');
+        }
+        if (!req.rpc_params.etag) {
+            throw new RpcError('INVALID_REQUEST', 'etag must be provided when using complete_upload');
+        }
+        info.etag = req.rpc_params.etag;
+        if (req.bucket.namespace && req.bucket.namespace.caching) {
+            info.cache_last_valid_time = new Date();
+        }
+    } else {
+        info.upload_size = 0;
+        info.upload_started = obj_id;
+        const lock_settings = config.WORM_ENABLED ? calc_retention(req) : undefined;
+        if (lock_settings) info.lock_settings = lock_settings;
```

Comment:
Same here - I suggest to pull this out of the complete_upload condition - if we decide to disable WORM settings for namespace cache buckets we can be more precise and do that in the namespace_cache level.

Reply:
ok